chain/src/pipe.rs

```diff
@@ -108,7 +108,49 @@ pub fn process_block(b: &Block, mut ctx: BlockContext) -> Result<Option<Tip>, Er
 			extension.force_rollback();
 		}
 		Ok(h)
-	})
+	});
+
+	match result {
+		Ok(t) => {
+			sync_file_metadata(&b, &sumtrees, ctx.store.clone())?; 
+			Ok(t)
+		},
+		Err(e) => Err(e),
+	}
+}
+
+fn sync_file_metadata(b: &Block, sumtrees: &sumtree::SumTrees, store: Arc<ChainStore>) -> Result<(), Error> {
+	// We want to store written file data for the last 2 blocks, so we potentially
+	// have a couple of options to roll back to
+
+	// TODO: Delete all old blocks before doing this
```

Comment:
I think we can handle the deletion at the same time we do compaction. So we'd delete the old blocks, delete the metedata record, and compact the MMR regularly.

Reply:
Yes, so long as compaction happens fairly regularly.. just worried about leaving stale pairings around in the DB in the event of forks, and I don't think we actually have any store functions implemented yet to delete all records by their prefix (sure it's trivial, just haven't looked up the docs). Compaction's probably a good place to do this